adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -256,156 +250,157 @@ class ADAMContext(val sc: SparkContext) extends Serializable with Logging {
     val job = HadoopUtil.newJob(sc)
     val records = sc.newAPIHadoopFile(filePath, classOf[AnySAMInputFormat], classOf[LongWritable],
       classOf[SAMRecordWritable], ContextUtil.getConfiguration(job))
+    if (Metrics.isRecording) records.instrument() else records
     val samRecordConverter = new SAMRecordConverter
 
     records.map(p => samRecordConverter.convert(p._2.get, seqDict, readGroups))
   }
 
-  private def maybeLoadBam[U <: ADAMPredicate[AlignmentRecord]](
+  def loadBam[U <: ADAMPredicate[AlignmentRecord]](
     filePath: String,
-    predicate: Option[Class[U]] = None,
-    projection: Option[Schema] = None): Option[RDD[AlignmentRecord]] = {
+    predicate: Option[Class[U]] = None): RDD[AlignmentRecord] = {
 
-    if (filePath.endsWith(".bam") || filePath.endsWith(".sam")) {
-
-      if (projection.isDefined) {
-        log.warn("Projection is ignored when loading a BAM file")
-      }
+    val reads = adamBamLoad(filePath)
 
-      val reads = adamBamLoad(filePath)
-
-      Some(applyPredicate(reads, predicate))
-    } else
-      None
+    applyPredicate(reads, predicate)
   }
 
-  private def maybeLoadFastq[U <: ADAMPredicate[AlignmentRecord]](
+  def loadParquetAlignments[U <: ADAMPredicate[AlignmentRecord]](
     filePath: String,
     predicate: Option[Class[U]] = None,
-    projection: Option[Schema] = None): Option[RDD[AlignmentRecord]] = {
+    projection: Option[Schema] = None): RDD[AlignmentRecord] = {
+    adamLoad[AlignmentRecord, U](filePath, predicate, projection)
+  }
 
-    if (filePath.endsWith(".ifq")) {
+  def loadInterleavedFastq[U <: ADAMPredicate[AlignmentRecord]](
```

Comment:
The predicate is optional with a default of `None`; if the user doesn't want to use a predicate they just don't provide a predicate.